docs/reviewing.md

```diff
@@ -112,8 +112,10 @@ and dependencies are correct.
 
 When components are defined in the `package_info` in `conanfile.py` the following conditions are desired
 
-- use the `cmake_find_package` or `cmake_find_package_multi` generators in `test_package/conanfile.py`
-- corresponding call to `find_package()` with the components _explicitly_ used in `target_link_libraries`
+- use `cmake_find_package` if library has an [official](cmake.org/cmake/help/latest/manual/cmake-modules.7.html#find-modules) CMake module emulated in the recipe.
+- use `cmake_find_package_multi` if library provides official cmake config file emulated in the recipe. If there are more than one target, try to use all of
```

Comment:
how do I know that?

Reply:
have to check the recipe directly. we already use the cmake_official_config.cmake. The official modules we can find here: https://cmake.org/cmake/help/latest/manual/cmake-modules.7.html#find-modules

Reply:
https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html

Is also a good indicator

Reply:
@uilianries may you add some short instructions for newbiews, what should be done and how to check if library provides an official cmake config file? it's not obvious right now.

Reply:
It's not a silver bullet, but when you have to remove `cmake` folder in `lib` or `share` folder, library likely provides an official CMake config file.

Reply:
there might be some rules, e.g. if recipe contains any of:
```
tools.rmdir(os.path.join(self.package_folder), "lib", "cmake")
tools.rmdir(os.path.join(self.package_folder), "share", "cmake")
```
it should use `cmake_find_package[_multi]` generator, from your words, it sounds like a regularity.
another possible rule, if recipe contains any of:
```
.names["cmake_find_package"]
.names["cmake_find_package_multi"]
```
it should use `cmake_find_package[_multi]` generator.
is it accurate?

Reply:
yes, and I prefer the last rule.

there are no `tools.rmdir(os.path.join(self.package_folder), "share", "cmake")` in recipes, because we remove `share` folder directly.

Reply:
@uilianries I also can't find any information on what is the `cmake_official_config.cmake`. can you explain that one as well?

Reply:
precedence:
- `self.cpp_info.filenames["cmake_find_package_multi"]`
- `self.cpp_info.names["cmake_find_package_multi"]`
- recipe name

Reply:
@SpaceIm these might be set on components as well:
```
        self.cpp_info.components["headers"].names["cmake_find_package"] = "headers"
        self.cpp_info.components["headers"].names["cmake_find_package_multi"] = "headers"
```
global names/filenames aren't mandatory, if library name matches cmake name (?)

Reply:
yes, I though that you were talking about config file name

Anyway, I strongly advice to be explicit and not rely on recipe name inheritence.

Reply:
There is also `set_property("cmake_target_name", ...)` and `set_property("cmake_file_name", ...)` now

Reply:
But only for `CMakeDeps`?

Reply:
nope, old generators also can use `set_property` (they are wrapped for comaptibility).
so the full list of patterns hook may scan for:

- `.names["cmake_find_package"]`
- `.names["cmake_find_package_multi"]`
- `.filenames["cmake_find_package"]`
- `.filenames["cmake_find_package_multi"]`
- `set_property("cmake_target_name"`
- `set_property("cmake_file_name"`
- `tools.rmdir(os.path.join(self.package_folder), "lib", "cmake")`
- `tools.rmdir(os.path.join(self.package_folder), "share", "cmake")` (unlikely to be encountered in the wild)

recipe name, I an not sure if should be checked at all. this list is dynamic and changes from one version of CMake to another.